app/routers/google_connect.py

```diff
@@ -0,0 +1,261 @@
+from datetime import datetime
+from fastapi import Depends, APIRouter, Request
+from starlette.responses import RedirectResponse
+
+from google.auth.transport.requests import Request as google_request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+
+from app.database.models import Event, User, OAuthCredentials, UserEvent
+from app.dependencies import get_db, SessionLocal
+from app.config import CLIENT_SECRET_FILE
+from app.routers.event import create_event
+from loguru import logger
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+router = APIRouter(
+    prefix="/google",
+    tags=["sync"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/sync")
+async def google_sync(request: Request,
+                      session=Depends(get_db)) -> RedirectResponse:
+    '''Sync with Google - if user never synced with google this funcion will take
+    the user to a consent screen to use his google calendar data with the app.
+    '''
+
+    user = get_active_user(session)  # getting active user
+
+    # getting valid credentials
+    credentials = get_credentials(user=user, session=session)
+
+    if credentials is None:
+        # in case credentials is none, this is mean there isn't a client_secret
+        logger.error("GoogleSync isn't available - missing client_secret.json")
+
+    # fetch and save the events com from Google Calendar
+    fetch_save_events(credentials=credentials, user=user, session=session)
+
+    return RedirectResponse(url=request.headers['referer'])
+
+
+def get_credentials(user: User,
```

Comment:
I think all the function from here to the end of the file should be in `app/internal`

Reply:
Done.